compatibility/basic-attestation-on-localhost-api-version-bump/test.sh

```diff
@@ -0,0 +1,112 @@
+[ -n "${RUST_KEYLIME_UPSTREAM_BRANCH}" ] || RUST_KEYLIME_UPSTREAM_BRANCH="master"
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        rlRun 'rlImport "./test-helpers"' || rlDie "cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+
+        # update /etc/keylime.conf
+        limeBackupConfig
+        rlRun "limeUpdateConf logger_root level DEBUG"
+        rlRun "limeUpdateConf logger_keylime level DEBUG"
+        rlRun "limeUpdateConf handler_consoleHandler level DEBUG"
+        # verifier
+        rlRun "limeUpdateConf revocations enabled_revocation_notifications '[]'"
+        # tenant
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        # agent
+        rlRun "limeUpdateConf agent enable_revocation_notifications false"
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+            # start ima emulator
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        # start keylime_verifier
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+        # create allowlist and excludelist
+        rlRun "limeCreateTestPolicy ${TESTDIR}/*"
+
+        WORKDIR=$( mktemp -d -p "/var/tmp" )
+    rlPhaseEnd
+
+    rlPhaseStartTest "Compile old and new keylime agent"
+        rlRun "git clone ${RUST_KEYLIME_UPSTREAM_URL} ${WORKDIR}/rust-keylime"
+        rlRun "pushd ${WORKDIR}/rust-keylime"
+        rlRun "git checkout v0.2.1"
+        rlRun "cargo build && cp ./target/debug/keylime_agent ${WORKDIR}/keylime_agent-2.1"
+        rlRun "git checkout ${RUST_KEYLIME_UPSTREAM_BRANCH}"
+        rlRun "cargo build && cp ./target/debug/keylime_agent ${WORKDIR}/keylime_agent"
+        rlRun "popd"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Add keylime agent with old API version"
+        rlRun "RUST_LOG=keylime_agent=debug ${WORKDIR}/keylime_agent-2.1 >> $(limeAgentLogfile) 2>&1 &"
+        AGENT_PID=$!
+        rlRun "cat > script.expect <<_EOF
+set timeout 20
+spawn keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u $AGENT_ID --verify --runtime-policy policy.json --cert default -c add
+expect \"Please enter the password to decrypt your keystore:\"
+send \"keylime\n\"
+expect eof
+_EOF"
+        rlRun "expect script.expect"
+        rlRun "limeWaitForAgentStatus $AGENT_ID 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'$AGENT_ID'" $rlRun_LOG -E
+    rlPhaseEnd
+
+    rlPhaseStartTest "Verify that API version is automatically bumped"
+        rlRun "kill $AGENT_PID"
+        rlRun "RUST_LOG=keylime_agent=debug ${WORKDIR}/keylime_agent >> $(limeAgentLogfile) 2>&1 &"
```

Comment:
Can't we simply start the service?

Reply:
If we don't recompile again and use the installed keylime, then I think we can simply start the service